Month3/PixelGameEngine/AsteroidGame/AsteroidMain.cpp

```diff
@@ -0,0 +1,463 @@
+					{
+						float Angle1 = ((float)rand() / (float)RAND_MAX) * 6.283185f;
+						float Angle2 = ((float)rand() / (float)RAND_MAX) * 6.283185f;
+						NewAsteroids.push_back({ Asteroid.PosX, Asteroid.PosY,10.0f * sinf(Angle1),10.0f * cosf(Angle1),(int)Asteroid.Size >> 1, 0.0f });
+						NewAsteroids.push_back({ Asteroid.PosX, Asteroid.PosY,10.0f * sinf(Angle2),10.0f * cosf(Angle2),(int)Asteroid.Size >> 1, 0.0f });
+					}
+					Asteroid.PosX = -100;
+					Score += 100;
+				}
+			}
+		}
+	}
+
+
+
+	void DestroyBullets()
+	{
+		if (Bullets.size() > 0)
+		{
+			auto Destroy =
+				remove_if(Bullets.begin(), Bullets.end(), [&](SpaceObject BulletDestroy)
+					{
+						return (BulletDestroy.PosX < 1 || BulletDestroy.PosY < 1 || BulletDestroy.PosX >= ScreenWidth() || BulletDestroy.PosY >= ScreenHeight());
+					});
+			if (Destroy != Bullets.end())
+			{
+				Bullets.erase(Destroy);
+			}
+		}
+	}
+
+	void DestroyAsteroids()
+	{
+		if (VecAsteroid.size() > 0)
+		{
+			auto Destroy =
+				remove_if(VecAsteroid.begin(), VecAsteroid.end(), [&](SpaceObject AsteroidDestroy)
+					{
+						return (AsteroidDestroy.PosX < 0);
+					});
+			if (Destroy != VecAsteroid.end())
+			{
+				VecAsteroid.erase(Destroy);
+			}
+		}
+	}
+
+	bool Collition(float AsteroidCenterX, float AsteroidCenterY, float Radius, float PosX, float PosY)
+	{
+		return sqrt(((PosX - AsteroidCenterX) * (PosX - AsteroidCenterX)) + ((PosY - AsteroidCenterY) * (PosY - AsteroidCenterY))) < Radius;
+	}
+
+	void ShipCollition()
+	{
+		for (auto& Asteroid : VecAsteroid)
+		{
+			if (Collition(Asteroid.PosX, Asteroid.PosY, Asteroid.Size, Player.PosX, Player.PosY))
+			{
+				Player.BeDead = true;
+			}
+		}
+	}
+
+	void BeDead()
+	{
+		if (Player.BeDead == true)
+		{
+			VecAsteroid.clear();
+			Bullets.clear();
+			Level = 1;
+
+			Player.PosX = ScreenWidth() / 2.0f;
+			Player.PosY = ScreenHeight() / 2.0f;
+			Player.DisplacementX = 0.0f;
+			Player.DisplacementY = 0.0f;
+			Player.Angle = 0.0f;
+			Player.BeDead = false;
+			ModelShip =
+			{
+				{0.0f,-15.0f},
+				{-5.0f , +5.0f},
+				{+5.0f,+5.0f}
+			};
+
+		}
+	}
+
+
+	virtual bool OnUserCreate()
+	{
+		VecAsteroid.push_back({ 50.0f,50.0f,10.0f,-15.0f,(int)20,0.0f,false });
+		InitialCharacteristicsPlayer();
+		InitialCharateriticsAsteroid();
+		return true;
+	}
+
+	virtual bool OnUserUpdate(float ElapsedTime)
+	{
+
+		BackgroundColor();
+		if (WinGame())
+		{
+			if (GetKey(olc::Key::SPACE).bReleased)
+			{
+				return false;
+			}
+
+		}
+		else
+		{
+
+			UpdateAndDrawAsteroids(ElapsedTime);
+			//DrawAsteroids(ElapsedTime);
+
+			TurnLeft(ElapsedTime);
+			TurnRight(ElapsedTime);
+			Aceleration(ElapsedTime);
+			Reverse(ElapsedTime);
+			Velocity(ElapsedTime);
+			DrawShip();
+			ShotBullet(ElapsedTime);
+			UpdateAndDrawBullets(ElapsedTime);
+			DestroyBullets();
+			DestroyAsteroids();
+			HudDisplay();
+			ShipCollition();
+			BeDead();
+			AppendNewAsteroids();
+			LevelUp();
+
+
+		}
+		return true;
+	}
+
+
+
+
```

Comment:
Many unnecessary blank lines

Reply:
If these blank lines are to section off the class with main, I think a comment divider would do the job a lot better.